tools/mapmerge2/dmm.py

```diff
@@ -83,6 +83,14 @@ def reassign_bad_keys(self, bad_keys):
             # reassign the grid entries which used the old key
             self.grid[k] = bad_keys.get(v, v)
 
+    def remove_unused_keys(self, modified_keys = None):
+        unused_keys = list(set(modified_keys)) if modified_keys is not None else self.dictionary.keys()
+        for key in self.grid.values():
+            if key in unused_keys:
+                unused_keys.remove(key)
+        for key in unused_keys:
+            del self.dictionary[key]
```

Comment:
```suggestion
        for key in unused_keys:
            self.dictionary.pop(key, None)
```

This is safer and a bit faster